documentation/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,186 @@
+
+Similar to the customized policy, there are changes regarding how the custom HTTP client is configured as well. You can now use the `(armcore.ConnectionOptions).HTTPClient` option in `github.com/Azure/azure-sdk-for-go/sdk/armcore` module to use your own implementation of HTTP client and plug in what they need into the configuration.
+
+**In old version (`services/**/mgmt/**`)** you cannot customize the HTTP client.
+
+**In new version (`sdk/**/arm**`)**
+
+```go
+httpClient := NewYourOwnHTTPClient{}
+conn := armcore.NewConnection(credential, &armcore.ConnectionOptions{
+    HTTPClient: &httpClient,
+})
+```
+
+### Error Handling
+
+There are some minor changes in the error handling. 
+
+- The errors returned by the SDK now are always of type `runtime.ResponseError` in `github.com/Azure/azure-sdk-for-go/sdk/internal/runtime` package which implements the `HTTPResponse` interface and `NonRetriableError` interface from `github.com/Azure/azure-sdk-for-go/sdk/azcore` package.
+- When there is an error in the SDK request, in the old version (`services/**/mgmt/**`), the return value will all be non-nil, and you can get the raw HTTP response from the response value. In the new version (`sdk/**/arm**`), the first return value will be empty and you need to cast the error to `HTTPResponse` interface to get the raw HTTP response. When the request is successful and there is no error returned, you will need to get the raw HTTP response in `RawResponse` property of the first return value.
+
+**In old version (`services/**/mgmt/**`)**
+
+```go
+resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters)
+if err != nil {
+	log.Fatalf("Response code: %d", resp.Response.Response.StatusCode)
+}
+```
+
+**Equivalent in new version (`sdk/**/arm**`)**
+
+```go
+resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters, nil)
+if err != nil {
+	rawResponse := err.(azcore.HTTPResponse).RawResponse()
+	log.Fatalf("Response code: %d", rawResponse.StatusCode)
+}
+```
+
+**When there is no errors in new version (`sdk/**/arm**`)**
+
+```go
+resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters, nil)
+if err != nil {
+	rawResponse := err.(azcore.HTTPResponse).RawResponse()
+	log.Fatalf("Response code: %d", rawResponse.StatusCode)
+}
+log.Printf("Response code: %d", resp.RawResponse.StatusCode)
+```
+
+### Long Running Operations
+
+In old version, if a request is a long-running operation, a struct `**Future` will be returned, which is an extension of the interface `azure.FutureAPI`. You need to invoke the `future.WaitForCompletionRef` to wait until it finishes.
+
+In the new version, if a request is a long-running operation, the function name will start with `Begin` to indicate this function will return an interface `**Poller` will be returned, which is an extension of the interface `azcore.Poller`.
```

Comment:
LROs don't return an interface in track 2, they return a response envelope that contains the poller.

Reply:
Done